kafka-streams-partitioners/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/RemoteServiceBasedConfigProvider.java

```diff
@@ -0,0 +1,56 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.time.Duration;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.partitioner.config.service.v1.GetPartitionerProfileRequest;
+import org.hypertrace.partitioner.config.service.v1.PartitionerConfigServiceGrpc.PartitionerConfigServiceBlockingStub;
+
+@Slf4j
+public class RemoteServiceBasedConfigProvider implements ConfigProvider {
```

Comment:
naming: it sounds more abstract than it is - it's a specific remote service so maybe just name it off that? Something like `PartitionerConfigServiceRemoteConfigProvider`? A bit wordy though, up to you.

Reply:
RemoteServiceBasedConfigProvider -> PartitionerConfigServiceCachingClient
This also bit wordy but more meaningful and inline with other config service clients.